sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -98,7 +100,12 @@ case class ExplainCommand(
 
   // Run through the optimizer to generate the physical plan.
   override def run(sparkSession: SparkSession): Seq[Row] = try {
-    val queryExecution = sparkSession.sessionState.executePlan(logicalPlan)
+    val queryExecution =
+      if (logicalPlan.isStreaming) {
+        new IncrementalExecution(sparkSession, logicalPlan, OutputMode.Append(), "dummy", 0)
```

Comment:
Will the mode affect the generated plan? Is it possible that the plan create at here is different from what the user expects?